### DISCUSSION TITLE
Separation of Scenes and Clips

### DISCUSSION THREAD

**Discussion by SpedNSFW** [stashapp/stash#2989] [Ideas]
As sites like OnlyFans and Fansly get more popular, so too are the amount of "clips" that become available. I would love to see a separation of short form content like clips where they are moved to a different page. As an example, look at YouTube. They have separated their main content from what they call "Shorts".

The reason I'm bringing this forward is because creators on the aforementioned platforms release clips by the dozen and very rarely have a proper name, release date, etc. I'm currently excluding any OF clips from being imported into my Stash instance because of all the clutter they cause but with what I'm proposing, I believe it could be a lot better.

Here is what I'm proposing:
 - Any video with a duration _shorter_ than a configurable value will be automatically designated as a **clip**.
 - Scenes will have an option added to mark them as a **clip**.
 - **Clips** will have an option to mark them as a scene.
 - The existing feature that exists to loop shorter scenes will be moved to **clips**. The duration option can remain as I can imagine that **clips** could be up to 5 minutes before one might classify it as a scene.
 - A mode, similar to lightbox slideshows for images, where after a **clip** is finished, the next will play. I think of stuff like Stories on platforms like Snapchat, YouTube, etc. I suppose this would work like the queue for scenes.

The way I imagine this being implemented is by having two classes, Scene and Clip, with a parent class (Video, for example). We could abstract a lot of the current stuff from scenes to the new parent class and have only what each of the children need in their own. I think for the most part they would display the same in the front end unless anyone has any ideas. I'm using OOP design talk here as it's easier for most people (including myself) to understand.

Even if this just ends up being a flag on scenes like "Organised" with invisible filtering behind the scenes, I think it would work nicely.

If anyone has seen any problems I've missed or has any ideas they'd like to add, please add your comment.

**Comment by philpw99** [comment]
The majority of videos are long ones. The handle methods and data structures are pretty much done. It's not wise to change them because short clips exist.
There are two major ways to organize your video files.
* If most of your videos are long ones, then one scene matches one video, and the scene contains all the information about that video.
* If most of your videos need to put into groups, like scene 1, scene 2, scene 3...etc., then you need to use "movies" to organize them and put the information on those movies.
* 
I use a mix of those two methods. For Japanese videos I put them in movies. For others I simply set the information on their scenes, and copy the info to the movie. I use my Stash_Helper to copy the scene info to a new movie easily.

So in your case, it seems you can use "movies" to organize the short clips too. Put the performer's name in the title of the movie, and you will see all the clips under that movie, and you can assign the order of clips anyway you want too.

**Comment by DogmaDragon** [comment]
For the most part, this is already possible with a simple Duration filter. You can also loop through them with the current queue system.
